### PR TITLE
fix(objects): compare and check versions correctness

### PR DESCRIPTION
**Pull request by JacopoDaeli**
Minor improvements to ensure versions correctness.

### CODE REVIEW

File: lib/plugins/object.js [hunk 1]
```diff
@@ -194,6 +195,77 @@ module.exports = fp(
...
+              ConditionExpression: '#a = :i and #b = :y',
+              ExpressionAttributeNames: {
+                '#a': 'headVersion',
+                '#b': 'latestVersion'
+              },
+              ExpressionAttributeValues: {
+                ':x': newHeadVersion,
+                ':z': newLatestVersion,
+                ':i': headVersion,
+                ':y': latestVersion
```
Comment (decompil3d) [inline lib/plugins/object.js:257]: Do these vars here and in values need to be single-character? Or can we use more meaningful names?

```suggestion
              UpdateExpression: 'set #headVersion = :newHeadVersion, #latestVersion = :newLatestVersion',
              ConditionExpression: '#headVersion = :headVersion and #latestVersion = :latestVersion',
              ExpressionAttributeNames: {
                '#headVersion': 'headVersion',
                '#latestVersion': 'latestVersion'
              },
              ExpressionAttributeValues: {
                ':newHeadVersion': newHeadVersion,
                ':newLatestVersion': newLatestVersion,
                ':headVersion': headVersion,
                ':latestVersion': latestVersion
```

Comment (JacopoDaeli) [reply lib/plugins/object.js:257]: The point of using single letters is to then benefit of `ExpressionAttributeNames` and `ExpressionAttributeValues` engine replacement. The suggestion you propose doesn't make much sense. In that case we could simply use the explicit name within the query string itself.

Comment (JacopoDaeli) [reply lib/plugins/object.js:257]: FYI, I am not against in not using `ExpressionAttributeNames`. I personally prefer the shorter syntax though.

Comment (JacopoDaeli) [reply lib/plugins/object.js]: Updated this by the way. I made things more explicit and removed `ExpressionAttributeNames`.

**Comment by JacopoDaeli** [comment]
@jcrugzz thoughts?

**Review by jcrugzz — APPROVED** [review]
Seems good to me 👍

**Review by decompil3d — APPROVED** [review]

**Review by ywangii — APPROVED** [review]

**Pull request merged**